src/executor.py

```diff
@@ -0,0 +1,171 @@
+    :param case: The case that is being executed.
+    :param block: The block that should be executed.
+    :param step: The step we are currently on in the case.
+    :return: Results from the block.
+    """
+    block_info = requests.get(block_url + block['name'] + '/info').json()
+
+    params = block['params']
+    insert_params = DotMap({'store': case['store'],
+                            'outputs': case['previous_outputs']})
+    for p_name, p_value in params.items():
+        if type(p_value) == str:
+            # Insert parameters from store and previous outputs.
+            params[p_name] = p_value.format_map(insert_params)
+
+    cleaned_params = {}
+    for p in block_info['params']:
+        try:
+            # Try to cast parameter to correct type.
+            # Required since all data from store or previous step are inserted as strings.
+            typ = type_map[block_info['params'][p]['type']]
+            cleaned_params[p] = typ(params[p])
+        except KeyError:
+            case_error(case, 'Case is missing parameter ' + p + ' required by the block ' +
+                       block['name'] + 'in step ' + step)
+            return None
+        except ValueError:
+            case_error(case, 'Failed to cast parameter value for parameter ' + p + ' in step ' + step)
+            return None
+
+    result = post_json(block['name'], {'params': cleaned_params})
+    return result
+
+
+def execute_case(case):
+    """
+    Handles the execution of a case, including the execution of single blocks and branching.
+    :param cid: The case ID.
+    """
+    try:
+        workflow = case['workflow']
+        step = case['step']
+
+        while True:
+            step_item = workflow['blocks'][step]
+
+            if step_item['type'] == 'action':
+                result = execute_block(case, step_item, step)
+
+                if not result:
+                    return
+
+                if result['type'] == 'result':
+                    case['previous_outputs'] = result['data']
+
+                    store = DotMap(case['store'])
+                    for saved_output, save_to in step_item['save_outputs'].items():
+                        if saved_output in result['data']:
+                            path = save_to.split('.')
+                            item = store
+                            for i in range(len(path) - 1):
+                                item = item[path[i]]
+                            item[path[-1]] = result['data'][saved_output]
+                    step = step_item['next_block']
+                    case['store'] = store.toDict()
+                elif result['type'] == 'suspend':
+                    case['suspended'] = True
+                    # TODO Need to save state, and handle the reason for suspension
+                    pass
+
+            if step == '-1':
+                case['status'] = CaseStatus.FINISHED
```

Comment:
This requirement for cases should be well documented

Reply:
Now documented in workflow-editor-service.